python/tutorials/09-persistent-matmul.py

```diff
@@ -373,46 +422,60 @@ def bench(K, reps=10):
     tma_persistent_result = matmul_tma_persistent(a, b)
 
-    naive_vs_cublas = "✅" if torch.allclose(naive_result.to(torch.float16), cublas_result.to(torch.float16),
-                                            atol=1.0) else "❌"
+    if torch_result is not None:
+        naive_vs_torch = "✅" if torch.allclose(naive_result.to(torch.float16), torch_result.to(torch.float16),
+                                               atol=1.0) else "❌"
+    if cublas_result is not None:
+        naive_vs_cublas = "✅" if torch.allclose(naive_result.to(torch.float16), cublas_result.to(torch.float16),
+                                                atol=1.0) else "❌"
     naive_vs_persistent = "✅" if torch.allclose(naive_result.to(torch.float16), persistent_result.to(torch.float16),
                                                 atol=1.0) else "❌"
     naive_vs_tma_persistent = "✅" if torch.allclose(cublas_result.to(torch.float16),
                                                     tma_persistent_result.to(torch.float16), atol=1.0) else "❌"
-    print(
-        f"M={M}, N={N}, K={K} verification naive vs: cublas {naive_vs_cublas}, persistent {naive_vs_persistent}, TMA persistent {naive_vs_tma_persistent}"
-    )
+    print(f"M={M}, N={N}, K={K} verification naive vs: ", end="")
+    if torch_result is not None:
+        print(f"torch: {naive_vs_torch} ", end="")
+    if cublas_result is not None:
+        print(f"cublas: {naive_vs_cublas} ", end="")
+    print(f"persistent: {naive_vs_persistent} ", end="")
+    print(f"TMA persistent: {naive_vs_tma_persistent}")
 
 
 if __name__ == "__main__":
-    if torch.cuda.is_available() and torch.cuda.get_device_capability()[0] >= 9:
-        parser = argparse.ArgumentParser()
-        parser.add_argument("-K", type=int, required=False)
-        parser.add_argument("--K_range", type=int, nargs=2)
-        parser.add_argument("--K_step", type=int, default=512)
-        args = parser.parse_args()
-
-        if args.K:
-            args.K_range = [args.K, args.K]
-            args.K_step = 1  # doesn't matter as long as it's not 0
-
-        torch.manual_seed(0)
-
-        validate(32, 32, 32)
-        validate(8192, 8192, 512)
-
-        proton.start("matmul", hook="triton")
-        for K in range(args.K_range[0], args.K_range[1] + 1, args.K_step):
-            bench(K)
-        proton.finalize()
-    else:
-        print("This tutorial fp8_matmul is only supported on CUDA with cc >= 90")
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-K", type=int, required=False)
+    parser.add_argument("--K_range", type=int, nargs=2)
+    parser.add_argument("--K_step", type=int, default=512)
+    parser.add_argument("--prec", type=str, choices=["fp8", "fp16"], default="fp8")
+    args = parser.parse_args()
+
+    if args.prec == 'fp8' and (not hasattr(torch, "float8_e4m3fn") or not is_cuda()):
+        print("This example requires CUDA with fp8 support.")
+        exit(1)
```

Comment:
@pawelszczerbuk I'm worried that using `exit` will cause troubles to the documentation pipeline.

Can you do a local test?

Reply:
```bash
          sudo pip3 install tabulate cmake sphinx matplotlib myst_parser sphinx-rtd-theme pandas pytest sphinx-gallery sphinx-multiversion
          cd docs
          export PATH=$(python3 -c "import cmake; print(cmake.CMAKE_BIN_DIR)"):$PATH
          python3 -m sphinx . _build/html/main
```

Reply:
Thanks Keren! The `exit(1)` was not a problem for the docs, however the fact that the script erred out if no arg was provided was! Fixed!